manager/src/lib.rs

```diff
@@ -531,157 +421,119 @@ impl CronManager {
         // get task based on current slot
         // priority goes to tasks that have fallen behind (using floor key)
         let mut slot_opt = self.slots.get(&current_slot);
-        let slot_ballpark = self.slots.floor_key(&current_slot);
-        let using_floor_key: bool = if let Some(k) = slot_ballpark {
-            slot_opt = self.slots.get(&k);
-            true
-        } else {
-            false
-        };
-
+        let mut slot_ballpark = None;
         if slot_opt.is_none() {
-            env::panic(b"No tasks found in slot");
+            slot_ballpark = self.slots.floor_key(&current_slot);
+            if let Some(k) = slot_ballpark {
+                slot_opt = self.slots.get(&k);
+            }
+        } else {
+            slot_ballpark = Some(current_slot);
         }
+
         let mut slot_data = slot_opt.unwrap();
```

Comment:
I am getting an error here when I run:
`cargo test simulate_many_tasks -- --nocapture`

```
ActionError(ActionError { index: Some(0), kind: FunctionCallError(HostError(GuestPanic { panic_msg: "panicked at \'called `Option::unwrap()` on a `None` value\', manager/src/lib.rs:434:38" })) })
```

Reply:
I believe this is now resolved from this commit: https://github.com/Cron-Near/contracts/pull/48/commits/9899f495d64300ffd11d3d0c771041f13c481e9f